csm_web/frontend/src/utils/types.js

```diff
@@ -1,9 +1,23 @@
 import PropTypes from "prop-types";
+export const OVERRIDE_SHAPE = PropTypes.shape({
+  date: PropTypes.string.isRequired,
+  // Hacky solution to avoid cyclic dependence between OVERRIDE_SHAPE and SPACETIME_SHAPE
```

Comment:
better solution would probably be to factor `dayOfWeek`, ..., `time` into a separate shape and pass it as a separate object, but that second layer of indirection would be annoying so this is probably fine as is

Reply:
Yeah, I'll leave this for now because I'm in a rush